test/e2e/docker_test.go

```diff
@@ -1334,3 +1334,15 @@ func TestDockerKubernetesRegionalCuratedPackages(t *testing.T) {
 		}
 	})
 }
+
+func TestDockerKubernetesUpgradeManagementComponents(t *testing.T) {
+	release := latestMinorRelease(t)
+	test := framework.NewClusterE2ETest(t, framework.NewDocker(t))
+
+	test.GenerateClusterConfig()
+	// create cluster with old eksa
+	test.CreateCluster(framework.ExecuteWithEksaRelease(release))
+	// upgrade management-components with new eksa
+	test.RunEKSA([]string{"upgrade", "management-components", "-f", test.ClusterConfigLocation, "-v", "99"})
```

Comment:
nit: I think that we should we create a `test.UpgradeManagmentComponents` function and hide these details.